### PR TITLE
Fixes a regression in forwarding from #6115

### PR DESCRIPTION
**Pull request by jefferai**
Although removing the authentication header is good defense in depth,
for forwarding mechanisms that use the raw request, we never add it
back. This caused perf standby tests to throw errors. Instead, once
we're past the point at which we would do any raw forwarding, but before
routing the request, remove the header.

To speed this up, a flag is set in the logical.Request to indicate where
the token is sourced from. That way we don't iterate through maps
unnecessarily.

### CODE REVIEW

File: http/handler.go [hunk 1]
```diff
@@ -687,6 +688,10 @@ func requestAuth(core *vault.Core, r *http.Request, req *logical.Request) (*logi
 	token, fromAuthzHeader := getTokenFromReq(r)
 	if token != "" {
 		req.ClientToken = token
+		req.ClientTokenSource = logical.ClientTokenFromVaultHeader
+		if fromAuthzHeader {
+			req.ClientTokenSource = logical.ClientTokenFromAuthzHeader
+		}
```
Comment (briankassouf) [inline http/handler.go]: Do we still want to delete on line 710 below?

Comment (jefferai) [reply http/handler.go]: Ah nope, good catch

File: vault/request_handling.go [hunk 2]
```diff
@@ -265,6 +265,23 @@ func (c *Core) checkToken(ctx context.Context, req *logical.Request, unauth bool
...
+	switch req.ClientTokenSource {
+	case logical.ClientTokenFromVaultHeader:
+		delete(req.Headers, consts.AuthHeaderName)
+	case logical.ClientTokenFromAuthzHeader:
+		if headers, ok := req.Headers["Authorization"]; ok {
+			for i, v := range headers {
+				if !strings.HasPrefix(v, "Bearer ") {
+					continue
+				}
+				headers = append(headers[:i], headers[i+1:]...)
```
Comment (briankassouf) [inline vault/request_handling.go]: I believe this will append non Bearer headers multiple times, see: https://play.golang.org/p/HJatGEPsjF7

Comment (jefferai) [reply vault/request_handling.go]: Good point. Just pushed a change, can you 👀 ?

Comment (briankassouf) [reply vault/request_handling.go]: Nice, looks better!

**Pull request merged**